pkg/test/framework/components/istio/operator.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"istio.io/istio/pkg/test/deployment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/core/image"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/scopes"
+)
+
+type operatorComponent struct {
+	id          resource.ID
+	settings    Config
+	ctx         resource.Context
+	environment *kube.Environment
+}
+
+var _ io.Closer = &operatorComponent{}
+var _ Instance = &operatorComponent{}
+var _ resource.Dumper = &operatorComponent{}
+
+// ID implements resource.Instance
+func (i *operatorComponent) ID() resource.ID {
+	return i.id
+}
+
+func (i *operatorComponent) Settings() Config {
+	return i.settings
+}
+
+func (i *operatorComponent) Close() (err error) {
+	scopes.CI.Infof("=== BEGIN: Cleanup Istio ===")
+	defer scopes.CI.Infof("=== DONE: Cleanup Istio ===")
+	if i.settings.DeployIstio {
```

Comment:
We should delete using the operator as well.
Currently the method is to dump (manifest generate) into a file and `kubectl delete -f` that file.

An `uninstall` method is being worked on for 1.5, in which we can use it.

Reply:
Does it work? That was the way helm was documented but the helm tests do a delete of the namespace, not sure if that was intentional. I guess testing the delete should be part of the test though

Reply:
yep. deletion must be tested as well.
At least you can run `delete` and ignore the errors from `kubectl`.
@ostromart could give the best advise on how to delete *currently*.